### PR TITLE
Writing tests for BookService

### PR DESCRIPTION
**Pull request by EnKarin**

### CODE REVIEW

File: src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import static org.assertj.core.api.Java6Assertions.assertThatThrownBy;
+
+class BookServiceTest extends BookCrossingBaseTests {
+
+    @Autowired
+    private BookService bookService;
+
+    @AfterEach
+    void delete() {
+        jdbcTemplate.update("delete from t_book where book_id in (1, 2, 3, 4)");
```
Comment (mfvanek) [inline src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java:27]: It won't work. You cannot bind to specific identifiers generated by the database in tests.

Comment (EnKarin) [reply src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java:27]: 2, 3 and 4 id set manually

File: src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="info">
+        <appender-ref ref="STDOUT"/>
+    </root>
+
+    <logger name="org.testcontainers" level="INFO"/>
```
Comment (mfvanek) [inline src/test/resources/logback-test.xml:12]: Do it via application-test.yml

File: src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                        null, 2022, null));
+    }
+
+    @SqlGroup({
+        @Sql("classpath:db/scripts/insert_user.sql"),
+        @Sql("classpath:db/scripts/insert_books.sql")
+    })
+    @Test
+    void findByTitleEmptyTest() {
+        assertThat(bookService.findByTitle("tit")).size().isEqualTo(0);
```
Comment (mfvanek) [inline src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java:200]:
```suggestion
        assertThat(bookService.findByTitle("tit"))
          .isEmpty();
```

File: src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    @SqlGroup({
+        @Sql("classpath:db/scripts/insert_user.sql"),
+        @Sql("classpath:db/scripts/insert_books.sql")
+    })
+    @Test
+    void findByTitleTest() {
+        final List<BookModelDto> returned = bookService.findByTitle("title");
+        assertThat(returned).size().isEqualTo(1);
```
Comment (mfvanek) [inline src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java:187]:
```suggestion
        assertThat(returned)
          .hasSize(1);
```

Comment (EnKarin) [reply src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java]: changed

File: src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java (hunk 3 above)
Comment (EnKarin) [reply src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java]: changed

**Pull request merged**